tenseal/tensors/bfvvector.cpp

```diff
@@ -178,23 +178,24 @@ BFVVector& BFVVector::mul_inplace(BFVVector to_mul) {
     return *this;
 }
 
-BFVVector BFVVector::mul_plain(vector<int64_t> to_mul) {
+BFVVector BFVVector::mul_plain(const vector<int64_t>& to_mul) {
     BFVVector new_vector = *this;
     new_vector.mul_plain_inplace(to_mul);
 
     return new_vector;
 }
 
-BFVVector& BFVVector::mul_plain_inplace(vector<int64_t> to_mul) {
+BFVVector& BFVVector::mul_plain_inplace(const vector<int64_t>& to_mul) {
     if (this->size() != to_mul.size()) {
         throw invalid_argument("can't multiply vectors of different sizes");
     }
 
     Plaintext plaintext;
+    vector<int64_t> new_vec_to_mul(to_mul);
     // prevent transparent ciphertext by adding a non-zero value
-    if (to_mul.size() + 1 <= this->context->slot_count<BatchEncoder>())
-        to_mul.push_back(1);
-    this->context->encode<BatchEncoder>(to_mul, plaintext);
+    if (new_vec_to_mul.size() + 1 <= this->context->slot_count<BatchEncoder>())
+        new_vec_to_mul.push_back(1);
+    this->context->encode<BatchEncoder>(new_vec_to_mul, plaintext);
```

Comment:
Can you add a TODO here to mention that we won't need to create a new vector when we remove the transparent hack so that we don't forget about it

Reply:
done